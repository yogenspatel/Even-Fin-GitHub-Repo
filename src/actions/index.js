/* eslint-disable no-console */
import Axios from "axios";

export const PERFORM_SEARCH = "PERFORM_SEARCH";
const GitSearchBaseURL = "https://api.github.com/search/repositories?q=";
export const LOADING_STATUS = {
  IN_PROGRESS: "IN_PROGRESS",
  ERROR: "ERROR",
  COMPLETE: "COMPLETE"
};
export const searchActions = {
  SET_SEARCH_STATUS: "SET_SEARCH_STATUS",
  SET_SEARCH_RESULTS: "SET_SEARCH_RESULTS",
  SET_SEARCH_ERROR: "SET_SEARCH_ERROR"
};

function setLoadingStatus(status) {
  return { type: searchActions.SET_SEARCH_STATUS, status };
}
function setSearchResults(results) {
  return {
    type: searchActions.SET_SEARCH_RESULTS,
    results
  };
}

function setError(error) {
  return {
    type: searchActions.SET_SEARCH_ERROR,
    error
  };
}

export function performSearch(queryParams) {
  const url = `${GitSearchBaseURL}${queryParams.searchText}${
    queryParams.stars ? `+stars:${queryParams.stars}` : ""
  }${queryParams.hasForked ? "+fork:true" : ""}+license:${
    queryParams.license ? queryParams.license : "mit"
  }`;
  const request = Axios.get(url);

  return {
    type: PERFORM_SEARCH,
    payload: request
  };
}

export function performSearchUsingThunk(queryParams) {
  const url = `${GitSearchBaseURL}${queryParams.searchText}${
    queryParams.stars ? `+stars:${queryParams.stars}` : ""
  }${queryParams.hasForked ? "+fork:true" : ""}+license:${
    queryParams.license ? queryParams.license : "mit"
  }`;

  return dispatch => {
    dispatch(setLoadingStatus(LOADING_STATUS.IN_PROGRESS));
    return Axios.get(url)
      .then(response => {
        dispatch(setLoadingStatus(LOADING_STATUS.COMPLETE));
        dispatch(setSearchResults(response));
      })
      .catch(error => {
        dispatch(setLoadingStatus(LOADING_STATUS.ERROR));
        dispatch(setError(error));
      });
  };
}
