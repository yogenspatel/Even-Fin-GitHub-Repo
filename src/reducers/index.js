import { createStore, combineReducers } from "redux";
import { reducer as reduxFormReducer } from "redux-form";
import searchReducer from "./search";

const rootReducer = combineReducers({
  form: reduxFormReducer, // mounted under "form"
  searchResults: searchReducer
});

function configureStore(state = {}) {
  return createStore(rootReducer, state);
}

export { configureStore };
export default rootReducer;
