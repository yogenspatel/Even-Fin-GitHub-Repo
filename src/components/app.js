import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { getUserData, LOADING_STATUS } from "../actions/index";
import Table from "./Table";
import Search from "./Search";
class App extends Component {
  componentDidMount() {
    this.props.getUserData();
  }
  render() {
    if (this.props.userData && this.props.userData.status === LOADING_STATUS.IN_PROGRESS) {
      return (
        <div>
          <h1>Loading...</h1>
        </div>
      );
    }
    if (this.props.userData && this.props.userData.status === LOADING_STATUS.ERROR) {
      return (
        <div>
          <h1>Error...</h1>
        </div>
      );
    }

    return (
      <div>
        <Search data={this.props.userData} />
        <Table />
      </div>
    );
  }
}

App.propTypes = {
  getUserData: PropTypes.func,
  userData: PropTypes.oneOfType([PropTypes.object, PropTypes.array])
};

App.defaultProps = {
  getUserData: () => {},
  userData: []
};

function mapStateToProps(state) {
  return {
    userData: state.userData
  };
}

export { App };

export default connect(
  mapStateToProps,
  { getUserData }
)(App);
