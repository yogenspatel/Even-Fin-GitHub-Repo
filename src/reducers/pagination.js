/* eslint-disable no-case-declarations */
import { PAGINATED_DATA } from "../actions";

export default (state = null, action) => {
  switch (action.type) {
    case PAGINATED_DATA:
      const { payload, noOfItems, pageSize, searchData } = action;
      return { ...state, paginated_data: payload, noOfItems, pageSize, searchData };
    default:
      return state;
  }
};
