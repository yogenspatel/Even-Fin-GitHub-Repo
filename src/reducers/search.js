import { PERFORM_SEARCH } from '../actions';

export default (state = null, action) => {
  switch (action.type) {
    case PERFORM_SEARCH:
      return action.payload.data.items;
    default:
      break;
  }
  return state;
};
