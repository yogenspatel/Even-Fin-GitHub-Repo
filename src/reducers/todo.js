import { GET_TODO_DATA, SET_LOADING_STATUS, SET_DATA_ERROR, SORT_DATA } from "../actions";

export default (state = null, action) => {
  switch (action.type) {
    case SET_LOADING_STATUS:
      return { ...state, status: action.status };
    case SET_DATA_ERROR:
      return { ...state, status: action.error };
    case GET_TODO_DATA:
      return action.payload ? action.payload : state;
    case SORT_DATA:
      console.log('state: ', action.sort_data);
      return action.sort_data;
    default:
      return state;
  }
};
