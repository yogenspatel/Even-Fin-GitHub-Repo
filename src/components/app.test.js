/* eslint-disable no-undef */

import React from "react";
import { App } from "./app";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { configureStore } from "../reducers";
import { LOADING_STATUS } from "../actions";

const data = [
  {
    id: 2,
    title: "this is test"
  },
  {
    id: 1,
    title: "delectus aut autem"
  },
  {
    id: 3,
    title: "quis ut nam facilis et officia qui"
  },
  {
    id: 4,
    title: "quo laboriosam deleniti aut qui"
  }
];

describe("App Component renders correctly", () => {
  it("App Component with Mock Data", () => {
    const store = configureStore();
    const tree = shallow(<App store={store} userData={data} />);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it("App Component without mock data", () => {
    const mockData = {
      status: LOADING_STATUS.IN_PROGRESS
    };
    const store = configureStore();
    const tree = shallow(<App store={store} userData={mockData} />);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it("App Component with error", () => {
    const mockData = {
      status: LOADING_STATUS.ERROR
    };
    const store = configureStore();
    const tree = shallow(<App store={store} userData={mockData} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
