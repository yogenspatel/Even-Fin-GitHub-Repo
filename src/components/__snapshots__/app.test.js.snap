// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders correctly App Component with Mock Data 1`] = `
<div>
  <Connect(Search)
    data={
      Array [
        Object {
          "id": 2,
          "title": "this is test",
        },
        Object {
          "id": 1,
          "title": "delectus aut autem",
        },
        Object {
          "id": 3,
          "title": "quis ut nam facilis et officia qui",
        },
        Object {
          "id": 4,
          "title": "quo laboriosam deleniti aut qui",
        },
      ]
    }
  />
  <Connect(Table) />
</div>
`;

exports[`App Component renders correctly App Component with error 1`] = `
<div>
  <h1>
    Error...
  </h1>
</div>
`;

exports[`App Component renders correctly App Component without mock data 1`] = `
<div>
  <h1>
    Loading...
  </h1>
</div>
`;
